### PR TITLE
Aaargh

### PR DESCRIPTION
**Pull request by clhm**
*Description of changes:*
Spelling, grammar, punctuation, and typos in first three chapters.
Phrasing adjustments in chapter 3.
Accumulator class edit. 

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_preliminaries/probability.md [hunk 1]
```diff
@@ -19,9 +19,9 @@ formal way of reasoning about our level of certainty.
 If we are completely sure
 that the image depicts a cat, we say that the *probability* that the corresponding label $y$ is "cat", denoted $P(y=$ "cat"$)$ equals $1$. 
 If we had no evidence to suggest that $y =$ "cat" or that $y =$ "dog", then we might say that the two possibilities were equally
-*likely* expressing this as $P(y=$ "cat"$) = P(y=$ "dog"$) = 0.5$. If we were reasonably
+*likely*, expressing this as <nobr>$P(y=$ "cat"$)$</nobr> $=$ <nobr>$P(y=$ "dog"$)$</nobr> $= 0.5$. If we were reasonably
```
Comment (astonzhang) [inline chapter_preliminaries/probability.md:22]: we won't use html tags

File: chapter_linear-networks/softmax-regression-gluon.md [hunk 2]
```diff
@@ -52,9 +52,8 @@ exponentiation can be a source of numerical stability issues
 (as discussed  in :numref:`sec_naive_bayes`).
 Recall that the softmax function calculates
 $\hat y_j = \frac{e^{z_j}}{\sum_{i=1}^{n} e^{z_i}}$, 
-where $\hat y_j$ is the $j^\mathrm{th}$ element of ``y_hat`` 
-and $z_j$ is the $j^\mathrm{th}$ element of the input
-``y_linear`` variable, as computed by the softmax.
+where $\hat y_j$ is the $j^\mathrm{th}$ element of $\hat{\mathbf{y}}$
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-gluon.md:55]: will keep the original version for now

File: chapter_preliminaries/probability.md [hunk 3]
```diff
@@ -19,9 +19,9 @@ formal way of reasoning about our level of certainty.
...
 confident, but not sure that the image depicted a cat, we might assign a
-probability $0.5  < P(y=$ "cat"$) < 1$.
+probability $0.5 <$ <nobr>$P(y=$ "cat"$)$</nobr> <nobr>$< 1$.</nobr>
```
Comment (astonzhang) [inline chapter_preliminaries/probability.md:24]: ditto

**Comment by astonzhang** [comment]
Thanks. Will merge and make some further edits based on your edits

**Comment by clhm** [comment]
Thanks.  I'll ignore line-break issues and avoid html fixes from now on.

**Pull request merged**